storages/backends/gcloud.py

```diff
@@ -0,0 +1,226 @@
+        return super(GoogleCloudFile, self).read(*args, **kwargs)
+
+    def write(self, content, *args, **kwargs):
+        if 'w' not in self._mode:
+            raise AttributeError("File was not opened in write mode.")
+        self._is_dirty = True
+        return super(GoogleCloudFile, self).write(force_bytes(content), *args, **kwargs)
+
+    def close(self):
+        if self._file is not None:
+            if self._is_dirty:
+                self.file.seek(0)
+                self.blob.upload_from_file(self.file)
+            self._file.close()
+            self._file = None
+
+
+@deconstructible
+class GoogleCloudStorage(Storage):
+    client_class = Client
+    file_class = GoogleCloudFile
+
+    not_found_exception = NotFound
+
+    project_id = setting('GS_PROJECT_ID', None)
+    credentials = setting('GS_CREDENTIALS', None)
+    bucket_name = setting('GS_BUCKET_NAME', None)
+    auto_create_bucket = setting('GS_AUTO_CREATE_BUCKET', False)
+    default_acl = setting('GS_DEFAULT_ACL', 'public-read')
+    bucket_acl = setting('GS_BUCKET_ACL', default_acl)
+    file_name_charset = setting('GS_FILE_NAME_CHARSET', 'utf-8')
+    file_overwrite = setting('GS_FILE_OVERWRITE', True)
+    # The max amount of memory a returned file can take up before being
+    # rolled over into a temporary file on disk. Default is 0: Do not roll over.
+    max_memory_size = setting('GS_MAX_MEMORY_SIZE', 0)
+
+    def __init__(self, **settings):
+        # check if some of the settings we've provided as class attributes
+        # need to be overwritten with values passed in here
+        for name, value in settings.items():
+            if hasattr(self, name):
+                setattr(self, name, value)
+
+        self._bucket = None
+        self._client = None
+
+    @property
+    def client(self):
+        if self._client is None:
+            self._client = self.client_class(
+                project=self.project_id,
+                credentials=self.credentials
+            )
+        return self._client
+
+    @property
+    def bucket(self):
+        if self._bucket is None:
+            self._bucket = self._get_or_create_bucket(self.bucket_name)
+        return self._bucket
+
+    def _get_or_create_bucket(self, name):
+        """
+        Retrieves a bucket if it exists, otherwise creates it.
+        """
+        try:
+            return self.client.get_bucket(name)
+        except self.not_found_exception:
+            if self.auto_create_bucket:
+                bucket = self.client.create_bucket(name)
+                bucket.acl.all().grant(self.bucket_acl)
+                bucket.acl.save()
+                return bucket
+            raise ImproperlyConfigured("Bucket %s does not exist. Buckets "
+                                       "can be automatically created by "
+                                       "setting GS_AUTO_CREATE_BUCKET to "
+                                       "``True``." % name)
+
+    def _clean_name(self, name):
```

Comment:
It doesn't need to be a method, no. I've changed it to just call `clean_name` directly.